### PR TITLE
[KOGITO-7752] Merge arrays should replace by default

### PR DESCRIPTION
**Pull request by fjtirado**
Before this JIRA, the code always try to merge the two arrays (an union), and the boolean just indicates if duplicates should be skipped. Now the boolean indicates whether the array should be merged (true) or replaced (false). Therefore true is equivalent to previous behaviour when skipDuplicate was true (this is the expected behaviour for JsonSchema generation), And by default (false), the array will be replaced, which is the desired behaviour for Serverless Workflow action

Why should be the default for Serveless Workflow?
Considers this case, where we have an array called minority in our workflow model with following content
```
[ {
  "message" : "Marc",
  "innerMessage" : {
    "number" : 23,
    "state" : "UNKNOWN"
  },
  "state" : "UNKNOWN"
}, {
  "message" : "Boungiorno Marco",
  "innerMessage" : {
    "number" : 23,
    "state" : "UNKNOWN"
  },
  "state" : "UNKNOWN"
} ]
```
And an expression in our workflow 
```

 {
       "name" : "expression",
       "type" : "expression",
       "operation" : ".minority[0].message = \"marquitos\""
    }
```
We want the result model to be 

```
[ {
  "message" : "marquitos",
  "innerMessage" : {
    "number" : 23,
    "state" : "UNKNOWN"
  },
  "state" : "UNKNOWN"
}, {
  "message" : "Boungiorno Marco",
  "innerMessage" : {
    "number" : 23,
    "state" : "UNKNOWN"
  },
  "state" : "UNKNOWN"
} ]
```

but if merge array is active (with skip duplicated) the result will be

```
[
{
  "message" : "marquitos",
  "innerMessage" : {
    "number" : 23,
    "state" : "UNKNOWN"
  },
  "state" : "UNKNOWN"
},
 {
  "message" : "Marc",
  "innerMessage" : {
    "number" : 23,
    "state" : "UNKNOWN"
  },
  "state" : "UNKNOWN"
}, {
  "message" : "Boungiorno Marco",
  "innerMessage" : {
    "number" : 23,
    "state" : "UNKNOWN"
  },
  "state" : "UNKNOWN"
} ]
```

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]

File: kogito-workitems/kogito-jackson-utils/src/main/java/org/kie/kogito/jackson/utils/MergeUtils.java [hunk 1]
```diff
@@ -36,43 +36,46 @@ public static JsonNode merge(JsonNode src, JsonNode target) {
...
-            return mergeArray(src, (ArrayNode) target, skipDuplicates);
+            return mergeArray(src, (ArrayNode) target, mergeArray);
         } else if (target.isObject()) {
-            return mergeObject(src, (ObjectNode) target, skipDuplicates);
+            return mergeObject(src, (ObjectNode) target, mergeArray);
         } else {
             if (src.isArray()) {
                 ArrayNode srcArray = (ArrayNode) src;
-                insert(srcArray, target, getExistingNodes(srcArray, skipDuplicates));
+                insert(srcArray, target, getExistingNodes(srcArray));
```
Comment (MarianMacik) [inline kogito-workitems/kogito-jackson-utils/src/main/java/org/kie/kogito/jackson/utils/MergeUtils.java:49]: If the boolean was false before, we would insert a duplicate here if it was already in the source array. Now we will always skip duplicates, regarding the boolean value. So there is a slight change in behaviour in this case, is that OK? 

So here we now discard duplicates always - when merging a src array with a target element.

File: kogito-workitems/kogito-jackson-utils/src/main/java/org/kie/kogito/jackson/utils/MergeUtils.java [hunk 2]
```diff
@@ -36,43 +36,46 @@ public static JsonNode merge(JsonNode src, JsonNode target) {
...
             if (src.isArray()) {
-                ((ArrayNode) src).forEach(node -> add(target, node, existingNodes));
+                if (mergeArray) {
+                    ((ArrayNode) src).forEach(node -> add(target, node, getExistingNodes(target)));
+                } else {
+                    return src;
+                }
             } else {
-                add(target, src, existingNodes);
+                add(target, src, Collections.emptySet());
```
Comment (MarianMacik) [inline kogito-workitems/kogito-jackson-utils/src/main/java/org/kie/kogito/jackson/utils/MergeUtils.java:78]: But here when we have the inverse situation - merging src element with target array - we will allow duplicates always.

Comment (fjtirado) [reply kogito-workitems/kogito-jackson-utils/src/main/java/org/kie/kogito/jackson/utils/MergeUtils.java:78]: hmmm, you are right (I was assuming src wont be there in the array, but this is not necessarily the case), let me fix this

**Review by MarianMacik — APPROVED** [review]
Thanks!

**Comment by kie-ci3** [comment]
**(tests) - kogito-runtimes job** `#640` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/640/display/redirect

**Test results:**
- PASSED: 2876
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/640/testReport/org.kie.kogito.quarkus.workflows/RPCGreetIT/testEnglish/">org.kie.kogito.quarkus.workflows.RPCGreetIT.testEnglish</a></summary>
java.lang.AssertionError: <br/>1 expectation failed.<br/>JSON path workflowdata.minority[0].message doesn't match.<br/>Expected: is "marquitos"<br/>  Actual: Bon diaMarc<br/><br/>	at org.kie.kogito.quarkus.workflows.RPCGreetIT.testEnglish(RPCGreetIT.java:46)<br/>
</details>

**Pull request merged**